scripts/data_generators/multidimensional_data_generation.py

```diff
@@ -0,0 +1,338 @@
+import numpy as np
+import pandas as pd
+import pymc as pm
+from pymc_extras.prior import Prior
+
+from pymc_marketing.mmm import GeometricAdstock, LogisticSaturation
+from pymc_marketing.mmm.multidimensional import MMM
+from pymc_marketing.paths import data_dir
+
+# global parameters
+
+seed = 12345
+geo_index = ("geo_a", "geo_b")
+chain = 0
+
+
+def get_dates(start_date="2022-06-06", end_date="2025-06-16"):
+    """Generate date range for the synthetic data.
+
+    Args:
+        start_date: Start date in YYYY-MM-DD format
+        end_date: End date in YYYY-MM-DD format
+
+    Returns:
+        dates: pandas DatetimeIndex with weekly dates
+        n_dates: number of dates
+
+    Raises:
+        ValueError: If end_date is before start_date or dates are invalid
```

Comment:
Can you please use NumPy doctring style (this is the one we use), see https://numpydoc.readthedocs.io/en/latest/format.html